src/app/gps/qgsgpstoolbar.cpp

```diff
@@ -151,14 +169,14 @@ QgsGpsToolBar::QgsGpsToolBar( QgsAppGpsConnection *connection, QgsMapCanvas *can
         mConnectAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/gpsicons/mIconGpsDisconnect.svg" ) ) );
         mConnectAction->setEnabled( true );
         mRecenterAction->setEnabled( true );
-        mAddFeatureAction->setEnabled( true );
+        mAddFeatureAction->setEnabled( static_cast< bool >( QgsProject::instance()->gpsSettings()->destinationLayer() ) );
```

Comment:
Should this be renamed to create feature to match the (good) renaming you did in QgsAppGpsDigitizing class?